packages/components/src/presentation-components/instance-filter-builder/PresentationInstanceFilterDialog.tsx

```diff
@@ -66,81 +68,95 @@ export function PresentationInstanceFilterDialog(props: PresentationInstanceFilt
-}
-
-function DelayLoadedPresentationInstanceFilterBuilder(props: DelayLoadedPresentationInstanceFilterBuilderProps) {
-  const { descriptorGetter, ...restProps } = props;
-  const descriptor = useDelayLoadedDescriptor(descriptorGetter);
-  if (!descriptor) {
-    return <DelayedCenteredProgressRadial />;
-  }
-
-  return <PresentationInstanceFilterBuilder {...restProps} descriptor={descriptor} />;
-}
-
-function useDelayLoadedDescriptor(descriptorGetter: () => Promise<Descriptor>) {
-  const [descriptor, setDescriptor] = useState<Descriptor>();
+function useDelayLoadedDescriptor(descr: Descriptor | (() => Promise<Descriptor>)) {
+  const [descriptor, setDescriptor] = useState<Descriptor | (() => Promise<Descriptor>)>(() => descr);
+  const disposed = useRef<boolean>(false);
 
   useEffect(() => {
-    let disposed = false;
     void (async () => {
-      const newDescriptor = await descriptorGetter();
-      // istanbul ignore else
-      if (!disposed) {
-        setDescriptor(newDescriptor);
+      if (descriptor && !(descriptor instanceof Descriptor)) {
+        const newDescriptor = await descriptor();
+        // istanbul ignore else
+        if (!disposed.current) {
+          setDescriptor(newDescriptor);
+        }
       }
     })();
     return () => {
-      disposed = true;
+      disposed.current = true;
     };
-  }, [descriptorGetter]);
+  }, [descr, descriptor]);
 
   return descriptor;
 }
 
-function DelayedCenteredProgressRadial() {
-  const [show, setShow] = useState(false);
+interface PresedntationInstanceFilterDialogContentProps extends Omit<PresentationInstanceFilterDialogProps, "isOpen" | "title" | "descriptor"> {
+  descriptor: Descriptor;
+}
 
-  useEffect(() => {
-    const timeout = setTimeout(() => {
-      setShow(true);
-    }, 250);
-    return () => {
-      clearTimeout(timeout);
-    };
-  }, []);
+function PresentationInstanceFilterDialogContent(props: PresedntationInstanceFilterDialogContentProps) {
+  const { onApply, initialFilter, descriptor, imodel, ruleGroupDepthLimit, filterResultCountRenderer, onClose } = props;
+
+  const { rootGroup, actions, validate } = usePropertyFilterBuilder({
+    initialFilter: initialFilter ? convertPresentationFilterToPropertyFilter(descriptor, initialFilter.filter) : undefined,
+  });
+
+  const filteringProps = usePresentationInstanceFilteringProps(descriptor, imodel, initialFilter?.usedClasses);
+
+  const navigationPropertyEditorContextValue = useFilterBuilderNavigationPropertyEditorContext(imodel, descriptor);
+
+  const applyButtonHandle = () => {
+    const filter = validate();
+    if (filter) {
+      const presentationInstanceFilter = createPresentationInstanceFilter(descriptor, filter);
+      if (presentationInstanceFilter) {
+        onApply({ filter: presentationInstanceFilter, usedClasses: filteringProps.selectedClasses });
+      }
+    }
+  };
+
+  const isDisabled = !rootGroup.items
+    .flat(ruleGroupDepthLimit ?? Number.MAX_SAFE_INTEGER)
```

Comment:
I don't think there is much difference between `flat(ruleGroupDepthLimit)` and `flat(Number.MAX_SAFE_INTEGER)`. So I think this can be simplified to `flat(Number.MAX_SAFE_INTEGER)`

Reply:
fixed